kamon-play-2.4.x/src/main/scala/kamon/play/instrumentation/instrumentation.scala

```diff
@@ -57,10 +59,10 @@ package object instrumentation {
   private def readOnlyTextMapFromHeaders(request: HttpRequest): TextMap = new TextMap {
     import scala.collection.JavaConverters._
 
-    private val headersMap = request.headers().iterator().asScala.map { h => h.getKey -> h.getValue }.toMap
+    private val headersMap = request.headers().iterator().asScala.map { h => h.getKey.toLowerCase(Locale.ENGLISH) -> h.getValue }.toMap
```

Comment:
It seems like there is not even need to have this `headersMap`, calling [get](https://github.com/playframework/playframework/blob/master/framework/src/play/src/main/scala/play/api/mvc/Headers.scala#L60) on the HttpHeaders should do the trick. The entire method could be as simple as:

```
  private def readOnlyTextMapFromHeaders(request: HttpRequest): TextMap = new TextMap {
    override def values: Iterator[(String, String)] = Iterator.empty
    override def get(key: String): Option[String] = Option(request.headers().get(key))
    override def put(key: String, value: String): Unit = {}
  }
```

Reply:
but the request.headers() refers to `org.jboss.netty.handler.codec.http.HttpHeaders` and not `play.api.mvc.Headers`

Reply:
Sorry, I tried that code locally and it worked, but linked to the wrong sources :/

Reply:
you are right